### PR TITLE
OCPBUGS-2968: bootstrap: Do not needlessly podman pull

### PR DESCRIPTION
**Pull request by celebdor**
When the release image is already present, the script should not try pulling. This slows down bootstrapping as well as breaking image pre-caching.

Signed-off-by: Antoni Segura Puimedon <antoni@redhat.com>

### CODE REVIEW

File: data/data/bootstrap/files/usr/local/bin/release-image-download.sh.template [hunk 1]
```diff
@@ -13,19 +13,21 @@ set -euo pipefail
...
-    fi
-done
+image_found=$(podman images --format json | jq --arg releaseimage "$RELEASE_IMAGE" 'any(.[].Names; .[] == $releaseimage)')
+if [[ "$image_found" == "false" ]]; then
+   echo "Pulling $RELEASE_IMAGE..."
+   while true; do
+       record_service_stage_start "pull-release-image"
+       if podman pull --quiet "$RELEASE_IMAGE"
+       then
+           record_service_stage_success
```
Comment (zaneb) [inline data/data/bootstrap/files/usr/local/bin/release-image-download.sh.template:23]: We probably want to still `record_service_stage_success` even if we don't need to pull an image.

File: data/data/bootstrap/files/usr/local/bin/release-image-download.sh.template [hunk 2]
```diff
@@ -13,19 +13,21 @@ set -euo pipefail
...
-    if podman pull --quiet "$RELEASE_IMAGE"
-    then
-        record_service_stage_success
-        break
-    else
-        record_service_stage_failure
-        echo "Pull failed. Retrying $RELEASE_IMAGE..."
-    fi
-done
+image_found=$(podman images --format json | jq --arg releaseimage "$RELEASE_IMAGE" 'any(.[].Names; .[] == $releaseimage)')
```
Comment (andfasano) [inline data/data/bootstrap/files/usr/local/bin/release-image-download.sh.template:16]: What about using `podman image exists "$RELEASE_IMAGE"`?

**Comment by r4f4** [comment]
@celebdor Are you still working on this PR? There are some reviewer comments to be addressed.

File: data/data/bootstrap/files/usr/local/bin/release-image-download.sh.template [hunk 3]
```diff
@@ -13,19 +13,23 @@ set -euo pipefail
...
-    then
-        record_service_stage_success
-        break
-    else
-        record_service_stage_failure
-        echo "Pull failed. Retrying $RELEASE_IMAGE..."
-    fi
-done
+if podman image exists "$RELEASE_IMAGE"; then
+   record_service_stage_start "pull-release-image"
```
Comment (andfasano) [inline data/data/bootstrap/files/usr/local/bin/release-image-download.sh.template:17]: nit: could be extracted before the if to remove the duplication

**Comment by r4f4** [comment]
/override ci/prow/golint
Linter issues are being fixed in https://github.com/openshift/installer/pull/6712 and this PR doesn't touch any Go files.

**Comment by bfournie** [comment]
/lgtm
Verified with dev-scripts

File: data/data/bootstrap/files/usr/local/bin/release-image-download.sh.template [hunk 4]
```diff
@@ -13,22 +13,27 @@ set -euo pipefail
...
-    else
-        printf '\n\\e{lightred}Unable to pull OpenShift release image\\e{reset}\n' >"${release_image_issue}"
-        record_service_stage_failure
-        echo "Pull failed. Retrying $RELEASE_IMAGE..."
-    fi
-done
+    record_service_stage_success
+else
+    echo "Pulling $RELEASE_IMAGE..."
+    release_image_issue="/etc/issue.d/50_release-image.issue"
```
Comment (zaneb) [inline data/data/bootstrap/files/usr/local/bin/release-image-download.sh.template:21]: This is also used on line 48, so it should be defined outside the if/else.

**Pull request merged**